crates/common/download/src/download.rs

```diff
@@ -450,111 +545,84 @@ mod tests {
         }
     }
 
-    // Parameters:
-    //
-    // status code
-    //
-    // bearer token boolean
-    //
-    // maybe url
-    //
-    // expected std error
-    //
-    // description
-    #[test_case(
-        200,
-        false,
-        Some("not_a_url"),
-        "builder error: relative URL without a base"
-        ; "builder error"
-    )]
-    #[test_case(
-        200,
-        true,
-        Some("not_a_url"),
-        "builder error: relative URL without a base"
-        ; "builder error with auth"
-    )]
-    #[test_case(
-        200,
-        false,
-        Some("http://not_a_url"),
-        "dns error: failed to lookup address information"
-        ; "dns error"
-    )]
-    #[test_case(
-        200,
-        true,
-        Some("http://not_a_url"),
-        "dns error: failed to lookup address information"
-        ; "dns error with auth"
-    )]
-    #[test_case(
-        404,
-        false,
-        None,
-        "404 Not Found"
-        ; "client error"
-    )]
-    #[test_case(
-        404,
-        true,
-        None,
-        "404 Not Found"
-        ; "client error with auth"
-    )]
     #[tokio::test]
-    async fn downloader_download_processing_error(
-        status_code: usize,
-        with_token: bool,
-        url: Option<&str>,
-        expected_err: &str,
-    ) -> anyhow::Result<()> {
-        let name = "test_download";
-        let version = Some("test1".to_string());
-        let target_dir_path = TempDir::new()?;
+    async fn resume_download_when_disconnected() -> anyhow::Result<()> {
+        env_logger::init();
+        let chunk_size = 4;
+        let file = "AAAABBBBCCCCDDDD";
+
+        let server_task = tokio::spawn(async move {
+            let listener = TcpListener::bind("localhost:3000").await.unwrap();
+
+            while let Ok((mut stream, _addr)) = listener.accept().await {
+                let response_task = tokio::time::timeout(Duration::from_secs(1), async move {
```

Comment:
What's this timeout needed for? To simulate the client disconnection?

Reply:
I originally added it to make sure that request/response terminates within 1s in the case that a client does not send data or sends it very slowly, but in a test this shouldn't matter at all, and only adds noise, so I'll be removing it.